Source/WTF/wtf/unix/UnixFileDescriptor.h

```diff
@@ -65,6 +67,16 @@ class UnixFileDescriptor {
         return *this;
     }
 
+    UnixFileDescriptor& operator=(const UnixFileDescriptor& o)
+    {
+        if (&o == this)
+            return *this;
+
+        this->~UnixFileDescriptor();
+        new (this) UnixFileDescriptor(o);
+        return *this;
```

Comment:
So it looks like you intend to duplicate the fd when it is copied, and I agree that's the only sensible way to implement a copy. But I don't think this actually works? In particular, the destructor is going to close the fd without setting m_value to 0. Then the constructor is going to attempt to dup() an invalid fd. Right?

So it needs to be reordered a bit, although you can't exactly reorder the constructor to run before the destructor....

Reply:
Yes, good catch. The destructor should set the value to -1.